ledger/src/blockstore_processor.rs

```diff
@@ -1360,13 +1360,18 @@ fn process_next_slots(
     blockstore: &Blockstore,
     leader_schedule_cache: &LeaderScheduleCache,
     pending_slots: &mut Vec<(SlotMeta, Bank, Hash)>,
+    halt_at_slot: Option<Slot>,
 ) -> result::Result<(), BlockstoreProcessorError> {
     if meta.next_slots.is_empty() {
         return Ok(());
     }
 
     // This is a fork point if there are multiple children, create a new child bank for each fork
     for next_slot in &meta.next_slots {
+        if halt_at_slot.is_some() && *next_slot > halt_at_slot.unwrap() {
```

Comment:
I'm guessing maybe:
`halt_at_slot.map(|halt_at_slot| *next_slot > halt_at_slot).unwrap_or_default()` would be preferred?
Maybe a version of if let Some(...), but that seems worse.
match ... seems worse, too.

Reply:
Updated, this is what I wanted to do but not in stable rust yet
```rust
let skip = if let Some(halt_at_slot) = halt_at_slot && halt_at_slot > *next_slot;
```

Reply:
Also, I knew `Option<T>` implemented `Eq` (`==, !=`), but it didn't occur to me that i could also support `PartialEq` (ie `>, <`). `None` appears to be treated as 0 but is distinct from 0; that is,
```
Some(4) > None  ==> true
Some(0) == None ==> false
```